tests/test_geoprocessing.py

```diff
@@ -285,8 +294,314 @@ def test_reproject_vector(self):
+        test_driver = ogr.GetDriverByName('ESRI Shapefile')
+        test_vector = test_driver.CreateDataSource(base_vector_path)
+        test_layer = test_vector.CreateLayer('base_layer', srs=base_srs)
+
+        for field_name in fields.keys():
+            field_defn = ogr.FieldDefn(field_name, ogr_types[field_name])
+            test_layer.CreateField(field_defn)
+        layer_defn = test_layer.GetLayerDefn()
+
+        test_feature = ogr.Feature(layer_defn)
+        test_geometry = ogr.CreateGeometryFromWkb(polygon_a.wkb)
+        test_feature.SetGeometry(test_geometry)
+        for field_name, field_val in fields.items():
+            test_feature.SetField(field_name, field_val)
+        test_layer.CreateFeature(test_feature)
+
+        test_layer = None
+        test_vector = None
+        test_driver = None
+
+        target_reference = osr.SpatialReference()
+        target_reference.ImportFromEPSG(26918) # UTM zone 18N
+
+        target_vector_path = os.path.join(
+            self.workspace_dir, 'target_vector.shp')
+        # create the file first so the model needs to deal with that
+        target_file = open(target_vector_path, 'w')
+        target_file.close()
+        pygeoprocessing.reproject_vector(
+            base_vector_path, target_reference.ExportToWkt(),
+            target_vector_path, layer_id=0, copy_fields=['id'])
+
+        vector = ogr.Open(target_vector_path)
+        layer = vector.GetLayer()
+        result_reference = layer.GetSpatialRef()
+        layer_defn = layer.GetLayerDefn()
+        self.assertTrue(
+            osr.SpatialReference(result_reference.ExportToWkt()).IsSame(
+                osr.SpatialReference(target_reference.ExportToWkt())))
+        self.assertTrue(layer_defn.GetFieldCount(), 1)
+        self.assertEqual(layer_defn.GetFieldIndex('id'), 0)
+
+        target_vector_no_fields_path = os.path.join(
+            self.workspace_dir, 'target_vector_no_fields.shp')
+        pygeoprocessing.reproject_vector(
+            base_vector_path, target_reference.ExportToWkt(),
+            target_vector_no_fields_path, layer_id=0, copy_fields=False)
+        layer = None
+        vector = None
+
+        vector = ogr.Open(target_vector_no_fields_path)
+        layer = vector.GetLayer()
+        result_reference = layer.GetSpatialRef()
+        layer_defn = layer.GetLayerDefn()
+        self.assertTrue(
+            osr.SpatialReference(result_reference.ExportToWkt()).IsSame(
+                osr.SpatialReference(target_reference.ExportToWkt())))
+        self.assertTrue(layer_defn.GetFieldCount(), 0)
         layer = None
         vector = None
+
+    def test_reproject_vector_latlon_to_utm(self):
+        """PGP.geoprocessing: reproject vector from lat/lon to utm."""
+        import pygeoprocessing
+        # Create polygon shapefile to reproject
+        base_srs = osr.SpatialReference()
+        base_srs.ImportFromEPSG(4326) # WGS84

```

Comment:
Can you add a space for the comment re: "PEP257, two spaces before inline # comment"?

Reply:
I think I got pep fixes in [f2c337b](https://github.com/natcap/pygeoprocessing/pull/31/commits/f2c337bce218e8c7f7d922edf0eb32f4d962f6df)